ql/experimental/callablebonds/callablebond.cpp

```diff
@@ -108,6 +109,270 @@ namespace QuantLib {
     }
 
 
+    /* Expose a mechanism to add spread the engine parameters and
+       clean clean up when the object is destroyed
+     */
+    class EngSpreadHelper {
+        CallableBond::arguments*  args;
+        double ospread;
+    public:
+        explicit EngSpreadHelper(CallableBond::arguments* args):
+            args(args),
+            ospread(args->spread)
+        {
+        }
+
+        void setSpread(double s)
+        {
+            args->spread=s;
+        }
+
+        ~EngSpreadHelper()
+        {
+            setSpread(ospread);
+        }
+    };
+
+    /* Functional interface to take a spread and return NPV of
+       spreaded callable bond */
+    class NPVSpreadHelper :
+        public std::unary_function<Real, Real>
+    {
+        CallableBond& bond_;
+        EngSpreadHelper &e;
+    public:
+        NPVSpreadHelper(CallableBond& bond,
+                        EngSpreadHelper& e):
+            bond_(bond),
+            e(e)
+        {
+        }
+        Real operator()(Real x) const
+        {
+            e.setSpread(x);
+            bond_.recalculate();
+            return bond_.NPV();
+        }
+    };
+
+    class OASHelper {
+    public:
+        OASHelper(const boost::function<Real(Real)>& npvhelper,
+                  Real targetValue):
+            npvhelper_(npvhelper),
+            targetValue_(targetValue)
+        {
+        }
+
+        Real operator()(Spread x) const
+        {
+            return targetValue_ - npvhelper_(x);
+        }
+    private:
+        const boost::function<Real(Real)>& npvhelper_;
+        Real targetValue_;
+    };
+
+    /* Convert a continuous spread to a conventional spread to a
+       reference yield curve
+    */
+    Real continuousToConv(Real oas,
+                          const Bond &b,
+                          RelinkableHandle<YieldTermStructure>& yts,
```

Comment:
If you don't mean to relink them (and you shouldn't) you can use `const Handle&` throughout.